src/core/namespace.h

```diff
@@ -93,11 +93,19 @@ struct TemporaryFileSystem {
         char *options;
 };
 
+typedef enum MountImageType {
+        MOUNT_IMAGE_DISCRETE,
+        MOUNT_IMAGE_OVERLAY,
+        _MOUNT_IMAGE_TYPE_MAX,
+        _MOUNT_IMAGE_TYPE_INVALID = -EINVAL,
+} MountImageType;
+
 struct MountImage {
         char *source;
         char *destination;
```

Comment:
hmm, so you are overloading this structure with a new purpose. which i guess is OK, given that setup_namespace() already is a ridiculous number of parameters and this is close enough. but please add a comment to the `char *destination` line, that clarifies the field is unsigned if MountImageType == MOUNT_IMAGE_OVERLAY.


(and one day we should probably move many of the parameters to setup_namespace() into NamespaceInfo)